tests/test_dcmip_setup.py

```diff
@@ -0,0 +1,161 @@
+
+    theta_b = Function(state.V[2])
+    theta_b.interpolate(Expression(theta_b_expr, a=a, G=G, T_eq=T_eq, u_0=u_0, N=N, g=g, p_eq=p_eq, R_d=R_d, kappa=kappa, p_0=p_0))
+
+    rho_b = Function(state.V[1])
+    rho_b.interpolate(Expression(rho_expr, a=a, G=G, T_eq=T_eq, u_0=u_0, N=N, g=g, p_eq=p_eq, R_d=R_d, kappa=kappa, p_0=p_0))
+
+    theta_prime = Function(state.V[2])
+    dis_expr = "a*acos(sin(phi_c)*sin(%(lat)s) + cos(phi_c)*cos(%(lat)s)*cos(%(lon)s - lamda_c))"
+
+    theta_prime_expr = "dT*(d*d/(d*d + pow((%(dis)s), 2)))*sin(2*pi*%%(z)s/L_z)" % {'dis': dis_expr} % string_expander
+
+    theta_prime.interpolate(Expression(theta_prime_expr, dT=deltaTheta, d=d, L_z=L_z, a=a, phi_c=phi_c, lamda_c=lamda_c))
+
+    theta0.assign(theta_b + theta_prime)
+    rho0.assign(rho_b)
+
+    state.initialise(u0, rho0, theta0)
+    state.set_reference_profiles(rho_b, theta_b)
+
+    # Set up advection schemes
+    advection_list = []
+    velocity_advection = NoAdvection(state)
+    advection_list.append((velocity_advection, 0))
+    rho_advection = LinearAdvection_V3(state, rho_b)
+    advection_list.append((rho_advection, 1))
+    theta_advection = LinearAdvection_Vt(state, k, theta_b)
+    advection_list.append((theta_advection, 2))
+    
+    # Set up linear solver
+    params = {'pc_type': 'fieldsplit',
+              'pc_fieldsplit_type': 'schur',
+              'ksp_type': 'gmres',
+              'ksp_monitor_true_residual': True,
+              'ksp_max_it': 100,
+              'ksp_gmres_restart': 50,
+              'pc_fieldsplit_schur_fact_type': 'FULL',
+              'pc_fieldsplit_schur_precondition': 'selfp',
+              'fieldsplit_0_ksp_type': 'richardson',
+              'fieldsplit_0_ksp_max_it': 2,
+              'fieldsplit_0_pc_type': 'bjacobi',
+              'fieldsplit_0_sub_pc_type': 'ilu',
+              'fieldsplit_1_ksp_type': 'richardson',
+              'fieldsplit_1_ksp_max_it': 2,
+              "fieldsplit_1_ksp_monitor_true_residual": True,
+              'fieldsplit_1_pc_type': 'gamg',
+              'fieldsplit_1_pc_gamg_sym_graph': True,
+              'fieldsplit_1_mg_levels_ksp_type': 'chebyshev',
+              'fieldsplit_1_mg_levels_ksp_chebyshev_estimate_eigenvalues': True,
+              'fieldsplit_1_mg_levels_ksp_chebyshev_estimate_eigenvalues_random': True,
+              'fieldsplit_1_mg_levels_ksp_max_it': 5,
+              'fieldsplit_1_mg_levels_pc_type': 'bjacobi',
+              'fieldsplit_1_mg_levels_sub_pc_type': 'ilu'}
+    
+    linear_solver = CompressibleSolver(state, alpha=0.5, params=params)
+
+    # Set up forcing
+    compressible_forcing = CompressibleForcing(state)
+
+    # build time stepper
+    stepper = Timestepper(state, advection_list, linear_solver,
+                          compressible_forcing)
+
+    return stepper, dt
+
+def run_dcmip():
+
+    stepper, dt = setup_dcmip()
+    stepper.run(t=0, tmax=dt)
+
+def test_dcmip_runs():
+
+    run_dcmip()
```

Comment:
Can you assert something about the state of the system when this has run?  Or is it only the case that you're testing for the stepper actually running?